eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/RelationConverter.php

```diff
@@ -66,30 +67,101 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
      */
     public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef)
     {
-        // Selection method, 0 = browse, 1 = dropdown
-        $storageDef->dataInt1 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionMethod'];
+        $fieldSettings = $fieldDef->fieldTypeConstraints->fieldSettings;
+        $doc = new DOMDocument('1.0', 'utf-8');
+        $root = $doc->createElement('related-objects');
+        $doc->appendChild($root);
 
-        // Selection root, location ID
-        $storageDef->dataInt2 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionRoot'];
+        $constraints = $doc->createElement('constraints');
+        if (!empty($fieldSettings['selectionContentTypes'])) {
+            foreach ($fieldSettings['selectionContentTypes'] as $typeIdentifier) {
+                $allowedClass = $doc->createElement('allowed-class');
+                $allowedClass->setAttribute('contentclass-identifier', $typeIdentifier);
```

Comment:
The transition should be easy as I did with the other fields (dataInt1 and dataInt2). We just need a new tag for the XML. Set the old tag's data as default if present (and delete it), and overwrite with the new tag's content if thats there. So later all xml's would only have the new tag.

The converter is a service so it should be easy to inject the content type service into it to resolve the ids. Just so many queries... content type service really should cache 😆 Or that multiple endpoint would be useful there.